invoicer_project/static/js/login_request.js

```diff
@@ -0,0 +1,105 @@
+const host = "http://127.0.0.1:8000";
+const emailForm = document.getElementById("email_input_log_in_page");
+const passwordForm = document.getElementById("password_input_log_in_page");
+
+
+function validatePassword() {
+    return passwordForm.value !== '';
+}
+
+function validateEmail() {
+    let user_email = emailForm.value;
+    return !(user_email.includes(' ') || !(/^[a-zA-Z0-9.]{3,20}@(?:[a-zA-Z0-9]{2,20}\.){1,30}[a-zA-Z]{2,10}$/.test(user_email))) 
+}
+
+function validateLoginDataOnFrontEnd() {
+    const emailRes = validateEmail();
+    const passwordRes = validatePassword();
+    return emailRes && passwordRes;
+}
+
+
+function visualEffects() {
+    emailForm.setAttribute('error', 'true');
+    passwordForm.setAttribute('error', 'true');
+    passwordForm.setAttribute('errorText', 'Invalid data');
+}
+
+function cancelVisualEffects() {
+    emailForm.removeAttribute('error');
+    passwordForm.removeAttribute('error');
+    passwordForm.removeAttribute('errorText');
+}
+
+
+
+document.getElementById("log_in_confirmation_button_log_in_page").addEventListener("click", async function (e) {
+    e.preventDefault();
+    if (validateLoginDataOnFrontEnd()) {
+        cancelVisualEffects();
+        const csrf_token = document.getElementsByName('csrfmiddlewaretoken')[0].value;
+        const email = emailForm.value;
+        const password = passwordForm.value;
+        const formData = new FormData();
+        formData.append('email', email);
+        formData.append('password', password);
+        formData.append('csrfmiddlewaretoken', csrf_token);
+
+        
+        const gotToken = await checkAndSaveTokens(host + '/user/authentication/', formData);
+        if (gotToken) {
+            window.location.href = host + '/clients/home/';
+        } 
+    } else {
+        visualEffects();
```

Comment:
I think there isn't enough context in names cancelVisualEffects and visualEffects
Let's discuss

Reply:
Maybe, VisualEffectsOnInputs and SettingErrorAttributesToInputs?

Reply:
Great suggestions!
SettingErrorAttributesToInputs looks seems ok

Reply:
done